### PR TITLE
Define occupationst and related structures

### PR DESCRIPTION
**Pull request by myk002**
and use it where it was referenced in existing structures as a comment

### CODE REVIEW

File: changelog.txt [hunk 1]
```diff
@@ -13,6 +13,10 @@ that repo.
 Template for new versions:
 
 ## Structures
+- ``Occupation``: enum defined
```
Comment (ab9rf) [inline changelog.txt:16]: i'd prefer that it be `occupation_type` as our C++ style doesn't allow enums to have caps-first names

File: df.art.xml [hunk 2]
```diff
@@ -1082,48 +1082,57 @@
         <enum-item name='MERCENARY'/>
```
Comment (ab9rf) [inline df.art.xml:1083]: add a comment to the `enum-type` reading `bay12: Occupation`

File: df.art.xml [hunk 3]
```diff
@@ -1082,48 +1082,57 @@
...
-        <int32_t name='unk_11'/>
-        <int32_t name='unk_12'/>
-        <int32_t name='unk_13'/>
-        <int32_t name='unk_14'/>
-        <int32_t name='unk_15'/>
-        <int32_t name='unk_16'/>
-        <int32_t name='unk_17'/>
-        <int32_t name='unk_18'/>
-        <int32_t name='unk_19'/>
+    <enum-type type-name='service_order_type'>
```
Comment (ab9rf) [inline df.art.xml:1109]: add a `comment` reading "bay12: ServiceOrder"

File: changelog.txt (hunk 1 above)
Comment (myk002) [reply changelog.txt:16]: done

**Review by ab9rf — CHANGES_REQUESTED** [review]

File: df.art.xml (hunk 3 above)
Comment (myk002) [reply df.art.xml]: done

File: df.art.xml (hunk 2 above)
Comment (myk002) [reply df.art.xml:1082]: done

**Review by ab9rf — APPROVED** [review]

File: df.art.xml [hunk 4]
```diff
@@ -1075,55 +1075,64 @@
...
+        <enum-item name='NONE' value='-1'/>
+        <enum-item name='DRINK'/>
+        <enum-item name='ROOM_RENTAL'/>
+        <enum-item name='EXTEND_ROOM_RENTAL'/>
+    </enum-type>
+
+    <struct-type type-name='service_orderst'>
+        <int32_t name='local_id'/>
+        <enum base-type='int32_t' name="type" type-name='service_order_type'/>
+        <pointer name='item_type' type-name='item_type'/>
```
[the remark this answers is not in the document]
Comment (myk002) [reply df.art.xml]: fixed. thanks!

**Pull request merged**